### PR TITLE
feat: add publish workflow when creating releases

### PR DESCRIPTION
**Pull request by Omar-Belghaouti**
Hi, I really liked your work here and I wanted to contribute to your project by adding a workflow for a GitHub Action that publish this package automatically to npm registry when creating releases in your repo. You just need to add `NODE_AUTH_TOKEN` to your repo secrets and you should test it before using it if any thing occurs.

### CODE REVIEW

**Comment by nexxeln** [comment]
Thank you! Would have appreciated if we could have discussed this in an issue first.

We're not making releases on GitHub yet, but this PR definitely made me want. I will not merge this for now, but we're gonna implement releasing on GitHub automatically soon and after that I will merge this.

**Comment by Omar-Belghaouti** [comment]
@nexxeln cool, glad to help!

File: .github/workflows/npm-publish.yml
Comment (jammaligad) [inline .github/workflows/npm-publish.yml:32]:
```suggestion
      - name: 🏗 build
```
small typo

File: .github/workflows/npm-publish.yml
Comment (JacobMGEvans) [inline .github/workflows/npm-publish.yml:21]: Not sure if this is from another repos Workflow, but there is a much newer version of this action
```suggestion
        uses: actions/setup-node@v3
```

File: .github/workflows/npm-publish.yml
Comment (JacobMGEvans) [inline .github/workflows/npm-publish.yml:18]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/npm-publish.yml
Comment (JacobMGEvans) [inline .github/workflows/npm-publish.yml:14]:
```suggestion
        node-version: [16.x]
```

Comment (JacobMGEvans) [reply .github/workflows/npm-publish.yml:14]: Oh nvm, this repo uses 14 or greater... Isn't 14 out of support?

Ok, it is maintained  
https://endoflife.date/nodejs

Comment (Omar-Belghaouti) [reply .github/workflows/npm-publish.yml]: It'll be out of maintenance by mid of 2023, but it is a good idea to use 16.x instead.

**Comment by Omar-Belghaouti** [comment]
@JacobMGEvans it's done

**Comment by juliusmarminge** [comment]
I think we should look into closing this @nexxeln @JacobMGEvans

**Comment by Omar-Belghaouti** [comment]
It can be applied using the input from the manual action dispatch:

```yml
on:
  workflow_dispatch:
    inputs:
      release-type:
        description: 'Release type (one of): patch, minor, major, prepatch, preminor, premajor, prerelease'
        required: true
```

**Comment by JacobMGEvans** [comment]
> I think we should look into merging this @nexxeln @JacobMGEvans
> 
> Is there a way to make this work with multiple tags? We have latest, next and beta.

Yes there is. You can see release Workflow here https://github.com/cloudflare/wrangler2/blob/main/.github/workflows/release.yml

You can see `beta` prerelease here https://github.com/cloudflare/wrangler2/blob/main/.github/workflows/prereleases.yml

**Review by amenahmyhrleart — APPROVED** [review]

**Comment by juliusmarminge** [comment]
Superseded by #308

**Pull request closed without merging**
